packages/api/src/external/ehr/athenahealth/command/get-patient.ts

```diff
@@ -0,0 +1,140 @@
+    baseUrl: athenaUrl,
+    patientId: athenaPatientId,
+  });
+  if (!athenaPatient) return undefined;
+  if (athenaPatient.name.length === 0) {
+    throw new Error("AthenaHealth patient missing at least one name");
+  }
+  if (athenaPatient.address.length === 0) {
+    throw new Error("AthenaHealth patient missing at least one address");
+  }
+
+  const patientDemos = createMetriportPatientDemos(athenaPatient);
+  const patients: Patient[] = [];
+  const getPatientByDemoWrapperErrors: string[] = [];
+
+  async function getPatientByDemoWrapper({
+    cxId,
+    demo,
+    patients,
+    errors,
+    log,
+  }: {
+    cxId: string;
+    demo: PatientDemoData;
+    patients: Patient[];
+    errors: string[];
+    log: typeof console.log;
+  }): Promise<void> {
+    try {
+      const patient = await getMetriportPatientByDemo({ cxId, demo });
+      if (patient) patients.push(patient);
+    } catch (error) {
+      const msg = `Failed to get patient by demo. Cause: ${errorToString(error)}`;
+      log(msg);
+      errors.push(msg);
+    }
+  }
+
+  await executeAsynchronously(
+    patientDemos.map(demo => {
+      return { cxId, demo, patients, errors: getPatientByDemoWrapperErrors, log };
+    }),
+    getPatientByDemoWrapper,
+    { numberOfParallelExecutions: 5 }
+  );
+
+  if (getPatientByDemoWrapperErrors.length > 0) {
+    capture.error("Failed to get patient by demo", {
+      extra: {
+        cxId,
+        patientCreateCount: patientDemos.length,
+        errorCount: getPatientByDemoWrapperErrors.length,
+        errors: getPatientByDemoWrapperErrors.join(","),
+        context: "athenahealth.get-patient",
+      },
+    });
+  }
+
+  const metriportPatient = patients[0];
+  if (metriportPatient) {
+    const uniquePatientIds = new Set(patients.map(patient => patient.id));
+    if (uniquePatientIds.size > 1) {
+      capture.message("AthenaHealth patient mapping to more than one Metriport patient", {
+        extra: {
+          cxId,
+          patientCreateCount: patientDemos.length,
+          patientIds: uniquePatientIds,
+          context: "athenahealth.get-patient",
+        },
+        level: "warning",
+      });
+    }
+    await findOrCreatePatientMapping({
+      patientId: metriportPatient.id,
+      externalId: athenaPatientId,
+      source: EhrSources.ATHENA,
+    });
+  }
+  return metriportPatient;
+}
+
+function createMetriportPatientDemos(patient: PatientResource): PatientDemoData[] {
+  const patientPatients: PatientDemoData[] = [];
+  const addressArray = createMetriportAddresses(patient);
+  const contactArray = createMetriportContacts(patient);
+  patient.name.map(name => {
+    const lastName = name.family;
+    name.given.map(firstName => {
+      patientPatients.push({
+        firstName,
+        lastName,
+        dob: normalizeDate(patient.birthDate),
+        genderAtBirth: normalizeGender(patient.gender),
+        address: addressArray,
+        contact: contactArray,
+      });
+    });
+  });
```

Comment:
Should we just join the first names separated by single space instead of mapping a single patient to multiple?

Reply:
Do we support that kind of look up for MPI?

Reply:
Looking here, seems like we don't loop through names to match on MPI, so separate patients may make sense.

https://github.com/metriport/metriport/blob/develop/packages/core/src/mpi/normalize-patient.ts#L22

Also, likely a bug but we only split by comma for names anyway (which is not in line with our docs). Same is true about the split function in domain/patient https://github.com/metriport/metriport/blob/5e2c64608a1855b5b82ff4c8e4657bb0a71de618/packages/core/src/domain/patient.ts#L88 (which ironically could be used in MPI anyway as it's in core too).

Feels like MPI normalization should use the new shared stuff from phone number + coverage assessment (and we should update patient demo augmentation as well).

Reply:
We chat about this and you made a good point that the resulting multiple demographics are just used as a filter for the get patient by demo. Also, that the alert is just in case the call to get patient by demo results in more than one patient. So all good here.